crates/router/src/core/payments/operations/payments_device_data_collection.rs

```diff
@@ -0,0 +1,310 @@
+                address: PaymentAddress {
+                    shipping: shipping_address.as_ref().map(|a| a.into()),
+                    billing: billing_address.as_ref().map(|a| a.into()),
+                },
+                confirm: Some(payment_attempt.confirm),
+                payment_attempt,
+                payment_method_data: None,
+                force_sync: None,
+                refunds: vec![],
+                disputes: vec![],
+                attempts: None,
+                sessions_token: vec![],
+                card_cvc: None,
+                creds_identifier: None,
+                pm_token: None,
+                connector_customer_id: None,
+                recurring_mandate_payment_data: None,
+                ephemeral_key: None,
+                multiple_capture_data: None,
+                redirect_response: None,
+                frm_message: None,
+                payment_link_data: None,
+                surcharge_details: None,
+            },
+            Some(customer_details),
+        ))
+    }
+}
+
+impl<F: Send + Clone, Ctx: PaymentMethodRetrieve>
+    ValidateRequest<F, api_models::payments::PaymentsDeviceDataCollectionRequest, Ctx>
+    for PaymentDeviceDataCollection
+{
+    #[instrument(skip_all)]
+    fn validate_request<'a, 'b>(
+        &'b self,
+        request: &api_models::payments::PaymentsDeviceDataCollectionRequest,
+        merchant_account: &'a domain::MerchantAccount,
+    ) -> RouterResult<(
+        BoxedOperation<'b, F, api_models::payments::PaymentsDeviceDataCollectionRequest, Ctx>,
+        operations::ValidateResult<'a>,
+    )> {
+        let request_merchant_id = Some(&request.merchant_id[..]);
+        helpers::validate_merchant_id(&merchant_account.merchant_id, request_merchant_id)
+            .change_context(errors::ApiErrorResponse::InvalidDataFormat {
+                field_name: "merchant_id".to_string(),
+                expected_format: "merchant_id from merchant account".to_string(),
+            })?;
+
+        let payment_id = request.payment_id.clone();
+
+        Ok((
+            Box::new(self),
+            operations::ValidateResult {
+                merchant_id: &merchant_account.merchant_id,
+                payment_id: api::PaymentIdType::PaymentIntentId(payment_id),
+                mandate_type: None,
+                storage_scheme: merchant_account.storage_scheme,
+                requeue: false,
+            },
+        ))
+    }
+}
+
+#[async_trait]
+impl<
+        F: Clone + Send,
+        Ctx: PaymentMethodRetrieve,
+        Op: Send + Sync + Operation<F, api_models::payments::PaymentsDeviceDataCollectionRequest, Ctx>,
+    > Domain<F, api_models::payments::PaymentsDeviceDataCollectionRequest, Ctx> for Op
+where
+    for<'a> &'a Op: Operation<F, api_models::payments::PaymentsDeviceDataCollectionRequest, Ctx>,
+{
+    #[instrument(skip_all)]
+    async fn get_or_create_customer_details<'a>(
```

Comment:
do we need to call `create_customer` flow here as it is only responsible for building the response HTML ?